src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -467,13 +470,15 @@ def generate_config(
         mgr_prometheus_sd_url = f'{srv_end_point}service=mgr-prometheus'  # always included
         ceph_exporter_sd_url = f'{srv_end_point}service=ceph-exporter'  # always included
         nvmeof_sd_url = f'{srv_end_point}service=nvmeof'  # always included
+        mgmt_gw_enabled = len(self.mgr.cache.get_daemons_by_service('mgmt-gateway')) > 0
 
         alertmanager_user, alertmanager_password = self.mgr._get_alertmanager_credentials()
         prometheus_user, prometheus_password = self.mgr._get_prometheus_credentials()
         FSID = self.mgr._cluster_fsid
 
         # generate the prometheus configuration
         context = {
+            'alertmanager_url_prefix': '/alertmanager' if mgmt_gw_enabled else '/',
```

Comment:
NIT: If this is at the end of the url (other places that did `/alertmanager`, `/prometheus` seemed to be) this is technically a suffix, not a prefix

Reply:
well, technically it's a prefix because it serves as prefix for any url you want to call on prometheus. Instead of calling: `http://<host>/my_url` now you should call `http://<host>/alertmanager/my_url` so it's a prefix for the url and that how the argument for pormetheus/alertmanager is called.